scripts/srv_update_sha256.sh

```diff
@@ -6,6 +6,9 @@
 # We only consider the earth_relief_xxy.grd files and the cache dir. SRTM tiles not included
 # If an error is detected we quit.
 
+# Name of hash table
+GMT_HASH_TABLE="gmt_hash_server"
```

Comment:
Are we going to use `gmt_hash_server.txt` or `gmt_data_server.txt`?

Reply:
No change, gmt_hash_server.txt since GMT 5/6.0 knows of no other name

Reply:
The above is about a new file that removes the hardwired information from gmt_datasets.h and gmt_remote.h into a live file.

Reply:
Is `gmt_data_server.txt` is manually updated? What about the data in the cache directory?

Reply:
gmt_data_server.txt is manually updated, just like the large data grids themselves.  They are not in git because of huge size.  The gmt_data_server.txt file will be in git of course and we will make new entries for new datasets we add.  I believe this is a relatively rare task.
The data in the cache directory is not really supported data sets, they are a hodge-podge of files we need for various examples and tests, and it makes it possible for others to run those examples and tests as well.  There is no guarantee taht any files in cache will be supported.  If anyone is using trench.txt from cache it may go away one day.

Reply:
Sorry, I'm still confused. 

We're going to have two hash tables in the data server, the old `gmt_hash_server.txt` and the new "gmt_data_server.txt". GMT<=6.0.0 only read the `gmt_hash_server.txt` file, and GMT>=6.1 only read the `gmt_data_server.txt`, right? If that's true, how does GMT6.1 know if a cache file is available?

Reply:
Here is the deal:

1. The gmt_hash_server.txt remains exactly as is.  Its job remains to inform GMT if a file should be updated (because there is a newer of of different size on the server) or if it should even be removed (because it is no longer on the server). All GMT version > 5.? knows about this file and its purpose and specific format remains the same.
2. GMT 6.1 will _in addition_ look for the gmt_data_server.txt file.  Its purpose is to provide information that we report the first time a file is downloaded (the attribution) plus it tells GMT what paths to build for the curl call.  The old system only knows about earth_relief and had a single directory for everything, except cache and srtm1|3.  Now, we may suddenly add a new dir earth_age and a bunch of files.  That info will go into the gmt_data_server.txt manually, and the size and hash info gets updated in the gmt_hash_server.txt automatically.

Is this clearer?

Reply:
Yes, good explanations. Now it sounds reasonable to me.